File: 0704-binary-search/0704-binary-search.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        if (nums.size() == 0){
            return -1;
        } 
        if (nums.size() == 1){
            return (nums[0] == target) -1;
        }
        int mid = nums.size()/2;
        if(nums[mid] == target){
            return mid;
        }
        if (nums[mid] < target){
            vector<int> newV = vector<int>(nums.begin() + mid, nums.begin() + nums.size());
            int right_search = search(newV, target); 
            return right_search < 0? right_search: mid + right_search; 
        }
        if (nums[mid] > target){
            vector<int> newV = vector<int>(nums.begin(), nums.begin()+mid);
            return search(newV, target);
            }
        return -2; //Shouldn't happen
        }
};

File: 0704-binary-search/0704-binary-search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        lo = 0
        hi = len(nums) - 1
        while(lo < hi):
            mid = lo + (hi - lo + 1)//2
            if nums[mid] == target:
                return mid
            if target > nums[mid]:
                lo = mid
            if target < nums[mid]:
                hi = mid - 1
        return -1 if nums[lo] != target else lo